primaryHandler.go

```diff
@@ -563,27 +563,34 @@ func validateDeviceID() alice.Chain {
 	})
 }
 
-func ValidateWRP() func(http.Handler) http.Handler {
+func ValidateWRP(logger *zap.Logger) func(http.Handler) http.Handler {
```

Comment:
we should get the logger from the request context instead

Reply:
i tried getting request form context but it doesn't exist in context at this point in time. the default logger that was being returned wasn't printing the statements.

Reply:
gotcha, oh that makes sense since it's 1 of the first middlewares